wp-includes/class-wp-https-ui.php

```diff
@@ -370,7 +370,7 @@ public function conditionally_redirect_to_https() {
 			if ( isset( $parsed_url['port'] ) ) {
 				$url .= ':' . $parsed_url['port'];
 			}
-			$url .= wp_unslash( $_SERVER['REQUEST_URI'] );
+			$url .= wp_unslash( sanitize_text_field( $_SERVER['REQUEST_URI'] ) );
```

Comment:
I don't think this should be done. Core doesn't do it, for example. You can search core for `wp_unslash.+?REQUEST_URI` and find many such examples:

```
src/wp-activate.php:23:list( $activate_path ) = explode( '?', wp_unslash( $_SERVER['REQUEST_URI'] ) );
src/wp-admin/widgets.php:400:					'return' => urlencode( remove_query_arg( wp_removable_query_args(), wp_unslash( $_SERVER['REQUEST_URI'] ) ) ),
src/wp-admin/edit-comments.php:129:	wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
src/wp-admin/network/user-new.php:85:				$edit_link = esc_url( add_query_arg( 'wp_http_referer', urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ), get_edit_user_link( $user_id_new ) ) );
src/wp-admin/nav-menus.php:622:						'return' => urlencode( remove_query_arg( wp_removable_query_args(), wp_unslash( $_SERVER['REQUEST_URI'] ) ) ),
src/wp-admin/includes/class-wp-users-list-table.php:445:			$edit_link = esc_url( add_query_arg( 'wp_http_referer', urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ), get_edit_user_link( $user_object->ID ) ) );
src/wp-admin/includes/theme.php:651:						'return' => urlencode( esc_url_raw( remove_query_arg( wp_removable_query_args(), wp_unslash( $_SERVER['REQUEST_URI'] ) ) ) ),
src/wp-admin/includes/class-wp-ms-users-list-table.php:273:			$edit_link = esc_url( add_query_arg( 'wp_http_referer', urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ), get_edit_user_link( $user->ID ) ) );
src/wp-admin/includes/class-wp-ms-users-list-table.php:504:			$edit_link       = esc_url( add_query_arg( 'wp_http_referer', urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ), get_edit_user_link( $user->ID ) ) );
src/wp-admin/includes/class-wp-ms-users-list-table.php:509:			$actions['delete'] = '<a href="' . esc_url( network_admin_url( add_query_arg( '_wp_http_referer', urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ), wp_nonce_url( 'users.php', 'deleteuser' ) . '&amp;action=deleteuser&amp;id=' . $user->ID ) ) ) . '" class="delete">' . __( 'Delete' ) . '</a>';
src/wp-admin/edit-tags.php:75:	$referer = wp_unslash( $_SERVER['REQUEST_URI'] );
src/wp-admin/edit-tags.php:212:	$location = remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) );
src/wp-admin/users.php:88:	$referer = '<input type="hidden" name="wp_http_referer" value="' . esc_attr( wp_unslash( $_SERVER['REQUEST_URI'] ) ) . '" />';
src/wp-admin/users.php:455:			wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
src/wp-admin/users.php:504:									urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ),
src/wp-admin/menu.php:190:	$customize_url            = add_query_arg( 'return', urlencode( remove_query_arg( wp_removable_query_args(), wp_unslash( $_SERVER['REQUEST_URI'] ) ) ), 'customize.php' );
src/wp-admin/upload.php:210:	wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
src/wp-admin/plugin-install.php:31:	$location = remove_query_arg( '_wp_http_referer', wp_unslash( $_SERVER['REQUEST_URI'] ) );
src/wp-admin/user-new.php:297:				$edit_link = esc_url( add_query_arg( 'wp_http_referer', urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ), get_edit_user_link( $user_id_new ) ) );
src/wp-admin/edit.php:211:	wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
src/wp-admin/admin.php:50:		wp_redirect( admin_url( 'upgrade.php?_wp_http_referer=' . urlencode( wp_unslash( $_SERVER['REQUEST_URI'] ) ) ) );
src/wp-admin/link-manager.php:43:	wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
src/wp-includes/functions.php:1838:	$referer_field = '<input type="hidden" name="_wp_http_referer" value="' . esc_attr( wp_unslash( $_SERVER['REQUEST_URI'] ) ) . '" />';
src/wp-includes/functions.php:1863:		$ref = 'previous' == $jump_back_to ? wp_get_referer() : wp_unslash( $_SERVER['REQUEST_URI'] );
src/wp-includes/functions.php:1888:	if ( $ref && wp_unslash( $_SERVER['REQUEST_URI'] ) !== $ref && home_url() . wp_unslash( $_SERVER['REQUEST_URI'] ) !== $ref ) {
src/wp-includes/feed.php:626:	return set_url_scheme( 'http://' . $host['host'] . wp_unslash( $_SERVER['REQUEST_URI'] ) );
src/wp-includes/class-wp-customize-manager.php:2096:		$self_url           = empty( $_SERVER['REQUEST_URI'] ) ? home_url( '/' ) : esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) );
src/wp-includes/cron.php:678:		wp_redirect( add_query_arg( 'doing_wp_cron', $doing_wp_cron, wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
src/wp-login.php:896:		list( $rp_path ) = explode( '?', wp_unslash( $_SERVER['REQUEST_URI'] ) );
```

So this should have the relevant sanitization rule ignored from PHPCS.